### PR TITLE
Update passcode decrypted by biometric key

### PR DESCRIPTION
**Pull request by biafra23**
Handles #1503

Changes proposed in this pull request:

@gnosis/mobile-devs

### CODE REVIEW

File: app/src/main/java/io/gnosis/safe/ui/settings/app/passcode/PasscodeViewModel.kt [hunk 1]
```diff
@@ -61,13 +61,6 @@ class PasscodeViewModel
                         updateState { PasscodeState(PasscodeCommandExecuted) }
                     }
                     BIOMETRICS_ENABLE -> {
-                        val cipher = biometricPasscodeManager.getInitializedRSACipherForEncryption(BiometricPasscodeManager.KEY_NAME)
```
Comment (biafra23) [inline app/src/main/java/io/gnosis/safe/ui/settings/app/passcode/PasscodeViewModel.kt:64]: not necessary here.

File: app/src/main/java/io/gnosis/safe/ui/settings/app/passcode/RepeatPasscodeFragment.kt [hunk 2]
```diff
@@ -115,6 +115,7 @@ class RepeatPasscodeFragment : BaseViewBindingFragment<FragmentPasscodeBinding>(
             input.doOnTextChanged(onSixDigitsHandler(digits, requireContext()) { digitsAsString ->
                 if (passcodeArg == digitsAsString) {
                     viewModel.setupPasscode(digitsAsString)
+                    viewModel.encryptPasscodeWithBiometricKey(digitsAsString)
```
Comment (biafra23) [inline app/src/main/java/io/gnosis/safe/ui/settings/app/passcode/RepeatPasscodeFragment.kt:118]: This is the fix. Update after setting the passcode

Comment (elgatovital) [reply app/src/main/java/io/gnosis/safe/ui/settings/app/passcode/RepeatPasscodeFragment.kt:118]: what happens if device doesn't support biometrics?

Comment (biafra23) [reply app/src/main/java/io/gnosis/safe/ui/settings/app/passcode/RepeatPasscodeFragment.kt:118]: This does not depend on the biometrics capability. But OTOH it is not necessary on devices that do not have biometrics functionality at all.

Comment (elgatovital) [reply app/src/main/java/io/gnosis/safe/ui/settings/app/passcode/RepeatPasscodeFragment.kt:118]: so check should be added to see if this action is necessary? Also this seems to be part of password setup. Why not combine it in a view model?

Comment (elgatovital) [reply app/src/main/java/io/gnosis/safe/ui/settings/app/passcode/RepeatPasscodeFragment.kt:118]: please add todo's, fixme's and let's revisit it later

**Review by elgatovital — APPROVED** [review]

**Pull request merged**